moviebot/nlu/nlu.py

```diff
@@ -36,17 +33,146 @@ def __init__(self, config):
         Args:
             config: Paths to ontology, database and tag words for slots in NLU.
         """
-        self.ontology: Ontology = config["ontology"]
-        self.database: DataBase = config["database"]
         self.intents_checker = UserIntentsChecker(config)
 
-    def generate_dact(  # noqa: C901
+    def _process_first_turn(
+        self, user_utterance: UserUtterance
+    ) -> List[DialogueAct]:
+        """Generates dialogue acts for the first turn of the conversation.
+
+        Args:
+            user_utterance: User utterance.
+
+        Returns:
+            A list of dialogue acts.
+        """
+        return (
+            self.intents_checker.check_reveal_voluntary_intent(user_utterance)
+            or self.intents_checker.check_basic_intent(
+                user_utterance, UserIntents.HI
+            )
+            or [DialogueAct(UserIntents.UNK, [])]
+        )
+
+    def _process_last_agent_dacts(
+        self, user_utterance: UserUtterance, last_agent_dacts: List[DialogueAct]
+    ) -> List[DialogueAct]:
+        """Processes response to agent dialogue acts from previous turn.
+
+        Args:
+            user_utterance: User utterance.
+            last_agent_dacts: Last agent dialogue acts.
+
+        Returns:
+            A list of dialogue acts.
```

Comment:
Can you specify in which case we are returning an empty list?

Reply:
Done